sdk/types/src/lib.rs

```diff
@@ -0,0 +1,59 @@
+#![no_std]
+
+use core::ops::Add;
+
+#[panic_handler]
+fn my_panic(_info: &core::panic::PanicInfo) -> ! {
+    loop {}
+}
+
+#[derive(Debug)]
+pub enum FeltError {
+    InvalidValue,
+}
+
+#[repr(transparent)]
+#[derive(Copy, Clone)]
+pub struct Felt(f64);
+
+impl Felt {
+    /// Field modulus = 2^64 - 2^32 + 1
+    const M: u64 = 0xFFFFFFFF00000001;
+
+    pub fn new(value: u64) -> Result<Self, FeltError> {
+        if value > Self::M {
+            Err(FeltError::InvalidValue)
+        } else {
+            Ok(Felt(value as f64))
+        }
+    }
+
+    pub fn as_u64(self) -> u64 {
+        self.0 as u64
+    }
+}
+
+impl From<Felt> for u64 {
```

Comment:
We should add `From<u8|u16|u32> for Felt`, as well as `From<usize> for Felt` (with a `#[cfg(target_pointer_width = "32")]` on it), since those are all trivially convertible to Felt, and likely to be common conversions.

Adding `TryFrom` for all of the signed versions of those values is probably worthwhile as well, but less important I think.

Reply:
Good catch! I added it in the next PR (where the proper `Felt` type is implemented) in https://github.com/0xPolygonMiden/compiler/pull/151/commits/bd3b5e732224252cfe97956363abb30a0b0e5c18